examples/getting-started/wp-templates/category.tsx

```diff
@@ -1,42 +1,90 @@
 import React from 'react';
 import { useGeneralSettings, usePosts } from '@wpengine/headless/react';
+import { Pagination } from '@wpengine/headless/next';
 import { getApolloClient, getPosts } from '@wpengine/headless';
 import { useRouter } from 'next/router';
 import { GetStaticPropsContext } from 'next';
 import { Footer, Header, Posts } from '../components';
 
-export default function Category(): JSX.Element {
-  const router = useRouter();
-  const urlParts = router.asPath.split('/').filter(Boolean);
-  const category = urlParts[1];
-  const posts = usePosts({
+const POSTS_PER_PAGE = 10;
+
+/**
+ * Parts of an optionally paginated category URL path such as
+ * /category/uncategorized/[before|after]/[abc123].
+ */
+interface CategoryUrlParts {
+  basePath: string; // '/category/uncategorized'
+  category: string; // 'uncategorized'
+  keyword?: string; // 'before'
+  id?: string; // 'abc123'
+}
+
+/**
+ * Converts a string or array into category URL parts assuming the
+ * form /category/uncategorized/[before|after]/[abc123].
+ */
+const getCategoryUrlParts = (path: string | string[]): CategoryUrlParts => {
```

Comment:
Is there anyway to bake `CategoryUrlParts`, `getCategoryUrlParts`, and `getQueryOptions` into the `Pagination` component itself? This is a lot of boilerplate. I would so love to just import `Pagination` from `@wpengine/headless` and literally use it with the framework just intrinsically knowing about the current cursor.

Reply:
Thanks for the feedback! I've refactored in https://github.com/wpengine/headless-framework/pull/154/commits/3b084b9bb8bec883510a6f2d6991562ecb014b46 to move the boilerplate out of the template. People don't need to know about pagination or cursors now.